modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/SearchAsYouTypeFieldMapper.java

```diff
@@ -736,8 +748,7 @@ protected void doXContentBody(XContentBuilder builder, boolean includeDefaults,
         List<Mapper> subIterators = new ArrayList<>();
         subIterators.add(prefixField);
         subIterators.addAll(Arrays.asList(shingleFields));
-        @SuppressWarnings("unchecked") Iterator<Mapper> concat = Iterators.concat(super.iterator(), subIterators.iterator());
-        return concat;
+        return Iterators.concat(super.iterator(), subIterators.iterator());
```

Comment:
It looks like we fail compilation on the warning from the call to `Iterators.concat` here

Reply:
yep I restored the original